### PR TITLE
Throw error if send unsupported type

### LINKED ISSUE #9: Throw error on send function object

### PR DESCRIPTION
**Pull request by VSerain**

### CODE REVIEW

File: src/HermesSerializers/index.js [hunk 1]
```diff
@@ -26,6 +26,9 @@ class HermesSerializers {
      * @param {any[]} args
      */
     serializeArgs(args) {
+        // Throw error is args is not sendable
```
Comment (VSerain) [inline src/HermesSerializers/index.js]: Throw error IF args is not sendable

File: src/HermesSerializers/index.js [hunk 2]
```diff
@@ -40,6 +43,21 @@ class HermesSerializers {
     unserializeArgs(args) {
         return this._serializers.reduce((args, serializer) => serializer.unserialize(args), args);
     }
+
+    checkPosibiltySend(object, path = "arguments") {
+        if (typeof object === "function") throw new Error(`You try to send a function in worker, this is not possible, please remove function at ${path}`);
```
Comment (VSerain) [inline src/HermesSerializers/index.js]: Remove "in worker"

File: src/HermesSerializers/index.js [hunk 3]
```diff
@@ -40,6 +43,21 @@ class HermesSerializers {
...
+        if (object instanceof Error) throw new Error(`You try to send Error in worker, this is not possible, please remove Error at ${path}`);
```
Comment (VSerain) [inline src/HermesSerializers/index.js]: Remove "in worker"

File: example/script.js [hunk 4]
```diff
@@ -57,4 +57,35 @@ window.onload = async () => {
...
+        await worker.call("log", [{
+            a: {
+                b: [
+                    0, 1, function test() {},
+                ],
+            },
+        }]);
+    } catch (e) {
+        console.log(e);
+    }
```
Comment (jbghoul) [inline example/script.js:90]: all those try-catches could be as one

**Review by jbghoul — CHANGES_REQUESTED** [review]

File: src/HermesSerializers/index.js (hunk 2 above)
Comment (robincartier) [reply src/HermesSerializers/index.js]: What about "Cannot send a function in worker, please remove it a ${path}" ?

File: example/script.js (hunk 4 above)
Comment (jbghoul) [reply example/script.js:90]: ah. no.

Comment (VSerain) [reply example/script.js:90]: Hello @jbghoul :D,
I am not sure, we want to show an error by example (One for Error, One for Function and One for HTMLElement).

If we added all example in one try catch we will have just one log.

But this part of repo is just for the example, we can make like you want, this part will soon evolve for real tests !

File: src/HermesSerializers/index.js (hunk 2 above)
Comment (VSerain) [reply src/HermesSerializers/index.js]: Yes it's good but, this error is throw also if worker try send function to page.

I suggest "Cannot send a function in other thread, please remove it a ${path}"

When do you say?

Comment (robincartier) [reply src/HermesSerializers/index.js]: "Worker cannot sent or receive any function, please remove it a ${path}" then ? 

You could also change the log for Error accordingly

Comment (VSerain) [reply src/HermesSerializers/index.js]: Yes ! 👍

Comment (Krenodeno) [reply src/HermesSerializers/index.js]: "Worker cannot send*"

Comment (Krenodeno) [reply src/HermesSerializers/index.js]: "remove it at*" ?

File: src/HermesSerializers/index.js (hunk 1 above)
Comment (VSerain) [reply src/HermesSerializers/index.js]: Done

File: src/HermesSerializers/index.js (hunk 2 above)
Comment (VSerain) [reply src/HermesSerializers/index.js]: @Krenodeno it's ok for u ?

File: src/HermesSerializers/index.js [hunk 5]
```diff
@@ -40,6 +43,21 @@ class HermesSerializers {
     unserializeArgs(args) {
         return this._serializers.reduce((args, serializer) => serializer.unserialize(args), args);
     }
+
+    checkPosibiltySend(object, path = "arguments") {
```
Comment (Krenodeno) [inline src/HermesSerializers/index.js]:
```suggestion
    checkPosibilitySend(object, path = "arguments") {
```

File: src/HermesSerializers/index.js (hunk 2 above)
Comment (Krenodeno) [reply src/HermesSerializers/index.js]: saw another typo, but for this one, it's all good ;)

File: src/HermesSerializers/index.js (hunk 5 above)
Comment (VSerain) [reply src/HermesSerializers/index.js]: Done

**Comment by Krenodeno** [comment]
LGTM

**Pull request merged**